lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java

```diff
@@ -215,6 +225,80 @@ private void sendIdentityIssuedAuditEvent(
                         extensions));
     }
 
+    private List<AuditEventReturnCode> getReturnCodes(
```

Comment:
Do you think we can simplify this if we pass the return codes from the user identity here? That way we don't need to duplicate that logic that determines the return codes themselves, and we can be sure that the return codes in the audit event match the user identity.

I'm imagining something like this in the audit event method
```
var auditEventReturnCodes = userIdentity.getReturnCode().stream()
        .map((returnCode -> getAuditEventReturnCode(returnCode, contraIndicators))).toList();
```

With this as the interesting part:
```
private AuditEventReturnCode getAuditEventReturnCode(
        ReturnCode returnCode,
        ContraIndicators contraIndicators
) {
    var ciConfig = configService.getContraIndicatorConfigMap();
    var issuers = contraIndicators.getContraIndicatorsMap().values().stream()
            .filter(ci -> ciConfig.get(ci.getCode()).getReturnCode().equals(returnCode.code()))
            .flatMap(ci -> ci.getIssuers().stream())
            .toList();
    return new AuditEventReturnCode(returnCode.code(), issuers);
}
```
(There might need to be some additional null-checking in there somewhere, but you get the idea...)

Reply:
This means we'll iterate through the config map once per return code, but in 99% of cases we only expect to have 0 or 1 return codes, so that seems fine!

Reply:
I think so, when trying to reach ContraIndicator through the method of reversing the return code, we cannot directly go from the CI.code. Because, If we want to access ContraIndicator from the return code, unfortunately, we should be working in the same business here like UserIndentityService.

Reply:
I'm not sure I follow - the `filter` line in the code above should match against CIs based on their return code (not CI code)?